src/components/Courses/components/CourseCard/CourseCard.tsx

```diff
@@ -17,17 +19,25 @@ type CourseCardProps = {
 
 const CourseCard: React.FC<CourseCardProps> = ({ course, authors }) => {
 	const navigate = useNavigate();
-	const dispatch = useDispatch();
+	const dispatch: AppDispatch = useDispatch();
+	const role = useSelector((state: RootState) => state.user.role);
 	const getAuthorsNames = (authorId: string[]): string => {
 		if (!authors) return '';
 
-		return authorId
-			.map((id) => authors.find((author) => author.id === id)?.name)
-			.join(', ');
+		return (
+			authorId &&
```

Comment:
Well actually here you are already checking for falsy authors so I think we can ommit authorId && check